applications/matter_weather_station/src/app_task.cpp

```diff
@@ -228,6 +244,10 @@ CHIP_ERROR AppTask::Init()
 	PrintOnboardingCodes(
 		chip::RendezvousInformationFlags(chip::RendezvousInformationFlag::kBLE));
 
+	gExampleDeviceInfoProvider.SetStorageDelegate(
```

Comment:
Shouldn't it be set before initializing the server? At least it would make sense to me to initialize all the providers before the server.